src/Runner.Worker/ExecutionContext.cs

```diff
@@ -411,21 +432,32 @@ public TaskResult Complete(TaskResult? result = null, string currentOperation =
                 this.Warning($"The job has experienced {TimeSpan.FromMilliseconds(_totalThrottlingDelayInMilliseconds).TotalSeconds} seconds total delay caused by server throttling.");
             }
 
+            DateTime now = DateTime.UtcNow;
             _record.CurrentOperation = currentOperation ?? _record.CurrentOperation;
             _record.ResultCode = resultCode ?? _record.ResultCode;
-            _record.FinishTime = DateTime.UtcNow;
+            _record.FinishTime = now;
             _record.PercentComplete = 100;
             _record.Result = _record.Result ?? TaskResult.Succeeded;
             _record.State = TimelineRecordState.Completed;
 
+            // Before our main timeline's final QueueTimelineRecordUpdate,
+            //    inject any issues collected by embedded ExecutionContexts.
+            if (!this.IsEmbedded)
+            {
+                foreach (var issue in _embeddedIssueCollector)
+                {
+                    AddIssue(issue);
+                }
+            }
+
             _jobServerQueue.QueueTimelineRecordUpdate(_mainTimelineId, _record);
 
             // complete all detail timeline records.
             if (_detailTimelineId != Guid.Empty && _detailRecords.Count > 0)
             {
                 foreach (var record in _detailRecords)
                 {
-                    record.Value.FinishTime = record.Value.FinishTime ?? DateTime.UtcNow;
+                    record.Value.FinishTime = record.Value.FinishTime ?? now;
```

Comment:
technically, we can remove all the code that deal with `_detailRecords` in a separate PR., those were from ADO and I am 90.999% sure we don't use them now.

Reply:
90.999% or 99.999% ?

Reply:
99.9999999999% 😄